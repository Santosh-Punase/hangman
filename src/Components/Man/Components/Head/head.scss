@import 'src/assets/scss/variables.scss';

.hangman__man {

  &-head {
    width: 6rem;
    height: 6rem;
    z-index: 10;
    position: relative;
  }

  &-face {
    width: 4rem;
    height: 6rem;
    margin: 0 1rem ;
    border-radius: 2rem;
    background-color: $brown;
    position: relative;
    overflow: hidden;
    z-index: 2;
  }

  &-ears {
    width: 5.5rem;
    height: 1rem;
    border-radius: 0.5rem;
    position: absolute;
    background-color: $yellow-light;
    top: 2.5rem;
    left: 0.25rem;

    &::after {
      position: absolute;
      content: '';
      width: 5rem;
      height: 0.75rem;
      background-color: $yellow-dark;
      border-radius: 1rem;
      top: 0.125rem;
      left: 0.25rem;
    }
  }

  &-forehead {
    width: 4rem;
    height: 3.5rem;
    margin: 0 1rem ;
    background-color: $yellow-light;
    position: absolute;
    top: 0;
    border-top-left-radius: 2rem;
    border-top-right-radius: 2rem;
    z-index: 3;
  }

  &-chick {
    background-color: $yellow-light;
    width: 1.75rem;
    height: 1rem;
    position: absolute;
    top: 2.75rem;
    z-index: 2;
    border-bottom-left-radius: 0.3rem;
    border-bottom-right-radius: 0.3rem;
    
    &.chick__left {
      left: 1rem;
      transform: rotate(-20deg);
      transform-origin: 100% 0;
    }

    &.chick__right {
      left: 3.25rem;
      transform: rotate(20deg);
      transform-origin: 0 0;
    }
  }

  &-nose {
    background-color: $yellow-light;
    width: 1rem;
    height: 1rem;
    position: absolute;
    top: 3.25rem;
    left: 2.5rem;
    z-index: 2;
    border-bottom-left-radius: 50%;
    border-bottom-right-radius: 50%;
  }

  &-mouth {
    background-color: $white;
    width: 1rem;
    height: 0.5rem;
    position: absolute;
    top: 4.5rem;
    left: 2.5rem;
    z-index: 2;
    border-top-left-radius: 1rem;
    border-top-right-radius: 1rem;
  }

  &-eyes {
    position: absolute;
    width: 0.5rem;
    height: 0.5rem;
    top: 2.75rem;
    background-color: $black;
    z-index: 3;
    border-radius: 50%;
    border: 1px solid black;

    &.eye__left {
      left: 1.75rem;
      &::before {
        position: absolute;
        content: '';
        width: 1rem;
        height: 0.5rem;
        top: -0.25rem;
        left: -0.25rem;
        border-top: 1px solid $gray-dark;
        border-radius: 40%;
      }
      &::after {
        position: absolute;
        content: '';
        width: 0.125rem;
        height: 0.125rem;
        left: 0.1rem;
        top: 0.1rem;
        background-color: $white;
        border-radius: 50%;
      }
    }

    &.eye__right {
      left: 3.75rem;
      &::before {
        position: absolute;
        content: '';
        width: 1rem;
        height: 0.5rem;
        top: -0.25rem;
        left: -0.25rem;
        border-top: 1px solid $gray-dark;
        border-radius: 40%;
      }
      &::after {
        position: absolute;
        content: '';
        width: 0.125rem;
        height: 0.125rem;
        left: 0.1rem;
        top: 0.1rem;
        background-color: $white;
        border-radius: 50%;
      }
    }
  }

  &-earing {
    position: absolute;
    width: 0.25rem;
    height: 0.25rem;
    top: 3.45rem;
    left: 5rem;
    border-radius: 50%;
    background-color: $blue-faded;
    z-index: 3;
  }
}