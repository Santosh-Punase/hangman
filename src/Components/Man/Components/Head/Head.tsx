import React from 'react';

import './head.scss';

const Head: React.FC = () => {
  
  return (
    <div className="hangman__man-head">
      <div className="hangman__man-ears"></div>
      <div className="hangman__man-face"></div>
      <div className="hangman__man-forehead"></div>
      <div className="hangman__man-eyes eye__left"></div>
      <div className="hangman__man-eyes eye__right"></div>
      <div className="hangman__man-chick chick__left"></div>
      <div className="hangman__man-chick chick__right"></div>
      <div className="hangman__man-nose"></div>
      <div className="hangman__man-mouth"></div>
      <div className="hangman__man-earing"></div>
    </div>

  );
}

export { Head };
