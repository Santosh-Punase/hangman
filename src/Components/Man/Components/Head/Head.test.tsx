import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { Head } from './Head';

afterEach(cleanup);

describe('Hangman Man Head', () => {
  test('should render head', () => {
    const { asFragment } = render(<Head />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
