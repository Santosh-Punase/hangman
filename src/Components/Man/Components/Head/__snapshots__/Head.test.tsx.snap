// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hangman Man Head should render head 1`] = `
<DocumentFragment>
  <div
    class="hangman__man-head"
  >
    <div
      class="hangman__man-ears"
    />
    <div
      class="hangman__man-face"
    />
    <div
      class="hangman__man-forehead"
    />
    <div
      class="hangman__man-eyes eye__left"
    />
    <div
      class="hangman__man-eyes eye__right"
    />
    <div
      class="hangman__man-chick chick__left"
    />
    <div
      class="hangman__man-chick chick__right"
    />
    <div
      class="hangman__man-nose"
    />
    <div
      class="hangman__man-mouth"
    />
    <div
      class="hangman__man-earing"
    />
  </div>
</DocumentFragment>
`;
